File: frontend/src/main/models/AnalysisResult.ts
import type {ParsedSection} from "./ParsedResume.ts";

export interface AnalysisResult {
    resumeId: string;
    sections: ParsedSection[];
    overallScore: number;
    atsCompatibility: number;
    keywordMatches: string[];
    missingKeywords: string[];
    recommendations: string[];
    createdAt: string;
}

File: backend/src/main/java/com/sammeydan/resumeanalyzer/backend/model/AnalysisResult.java
package com.sammeydan.resumeanalyzer.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AnalysisResult {
    private String resumeId;
    private List<ParsedSection> sections;
    private int overallScore;
    private int atsCompatibility;
    private List<String> keywordMatches;
    private List<String> missingKeywords;
    private List<String> recommendations;
    private LocalDateTime createdAt;
}
